driver-mapping/src/test/java/com/datastax/driver/mapping/PropertyNamingStrategyTest.java

```diff
@@ -0,0 +1,314 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.mapping;
+
+import com.datastax.driver.core.CCMTestsSupport;
+import com.datastax.driver.mapping.annotations.Column;
+import com.datastax.driver.mapping.annotations.PartitionKey;
+import com.datastax.driver.mapping.annotations.Table;
+import com.datastax.driver.mapping.configuration.MapperConfiguration;
+import com.datastax.driver.mapping.configuration.naming.CommonNamingConventions;
+import org.testng.annotations.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Test for JAVA-1316 - validate ability to automatically translate property names
+ * across different naming conventions.
+ */
+@SuppressWarnings("unused")
+public class PropertyNamingStrategyTest extends CCMTestsSupport {
```

Comment:
The following conventions seem untested: UpperCamelCase, UpperSnakeCase, LowerKebabCase, UpperKebabCase.

I have a mixed feeling about the "kebab" variants. 

First of all, this "kebab" name: as per [this SO question](http://stackoverflow.com/questions/11273282/whats-the-name-for-hyphen-separated-case/12273101) it seems we could call it "lisp-case" or "spinal-case" as well. If for nothing else, I'd rather pay hommage to a beautiful programming language than to some greasy – yet savory – meat skewers :)

Secondly, as you said, the kebab variants are not possible in Java. They _could_ be possible in Cassandra, provided that the column name is double-quoted: `"MY-COLUMN"` is a valid CQL identifier. This should be explicitly tested for.

I'd suggest here the use of a `@DataProvider` annotation that would test all the possible valid combinations of Java + Cassandra naming conventions.